src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileUtils.cs

```diff
@@ -295,5 +344,129 @@ private static bool GroupHasNonEmptyItems(IList<LockFileItem> group)
         {
             return group?.Any(item => !item.Path.EndsWith(PackagingCoreConstants.ForwardSlashEmptyFolder)) == true;
         }
+
+        /// <summary>
+        /// Group all items by the primary key, then select the nearest TxM 
+        /// within each group.
+        /// Items that do not contain the primaryKey will be filtered out.
+        /// </summary>
+        private static List<ContentItemGroup> GetContentGroupsForFramework(
+            LockFileTargetLibrary lockFileLib,
+            NuGetFramework framework,
+            List<ContentItemGroup> contentGroups,
+            string primaryKey)
+        {
+            var groups = new List<ContentItemGroup>();
+
+            // Group by primary key and find the nearest TxM under each.
+            var primaryGroups = new Dictionary<string, List<ContentItemGroup>>(StringComparer.Ordinal);
+
+            foreach (var group in contentGroups)
+            {
+                object keyObj;
+                if (group.Properties.TryGetValue(primaryKey, out keyObj))
+                {
+                    string key = (string)keyObj;
+
+                    List<ContentItemGroup> index;
+                    if (!primaryGroups.TryGetValue(key, out index))
+                    {
+                        index = new List<ContentItemGroup>(1);
+                        primaryGroups.Add(key, index);
+                    }
+
+                    index.Add(group);
+                }
+            }
+
+            // Find the nearest TxM within each primary key group.
+            foreach (var primaryGroup in primaryGroups)
+            {
+                var groupedItems = primaryGroup.Value;
+
+                var nearestGroup = NuGetFrameworkUtility.GetNearest<ContentItemGroup>(groupedItems, framework,
+                    group =>
+                    {
+                        // In the case of /native there is no TxM, here any should be used.
+                        object frameworkObj;
+                        if (group.Properties.TryGetValue(
+                            ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker,
+                            out frameworkObj))
+                        {
+                            return (NuGetFramework)frameworkObj;
+                        }
+
+                        return NuGetFramework.AnyFramework;
+                    });
+
+                // If a compatible group exists within the secondary key add it to the results
+                if (nearestGroup != null)
+                {
+                    groups.Add(nearestGroup);
+                }
+            }
+
+            return groups;
+        }
+
+        private static List<LockFileItem> GetRuntimeTargetLockFileItems(
+            RestoreTargetGraph targetGraph,
```

Comment:
Sorry, that maybe was a really non-specific comment. I don't see where the `RestoreTargetGraph targetGraph` parameter is used.